transaction/core/src/validation/validate.rs

```diff
@@ -86,6 +85,12 @@ pub fn validate<R: RngCore + CryptoRng>(
         validate_outputs_are_sorted(&tx.prefix)?;
     }
 
+    if block_version.signed_input_rules_are_supported() {
+        validate_all_input_rules(block_version, tx)?;
+    } else {
+        validate_no_input_rules_exist(tx)?;
```

Comment:
Consider renaming this method. I read it as "validate zero input rules" with an "exist" tacked on at the end. Maybe something like "validate_when_no_input_rules_exist" OR "validate_input_rules_not_supported"

Reply:
the other functions in this file are named like `validate_no_memos_exist` etc. but most of these were named by me.

i could change it to like `validate_that_no_memos_exist` ? does that seem better?